packages/lib/src/components/internal/PayButton/PayButton.tsx

```diff
@@ -10,27 +10,50 @@ export interface PayButtonProps {
      */
     classNameModifiers?: string[];
     label?: string;
-    amount: PaymentAmount;
+    amount: PaymentAmountExtended;
+    secondaryAmount?: PaymentAmountExtended;
     status?: string;
     disabled?: boolean;
     icon?: string;
     onClick?(): void;
 }
 
-const payAmountLabel = (i18n: Language, amount) =>
-    `${i18n.get('payButton')} ${!!amount?.value && !!amount?.currency ? i18n.amount(amount.value, amount.currency) : ''}`;
+const payAmountLabel = (i18n: Language, amount) => {
```

Comment:
Can you add the type `PaymentAmountExtended` for `amount`?

Reply:
can do!